src/glados/db.py

```diff
@@ -0,0 +1,290 @@
+import logging
+from datetime import datetime
+from typing import List, Optional
+from uuid import uuid4
+
+from sqlalchemy import Column, DateTime, Integer, MetaData, String, Table, create_engine
+from sqlalchemy.dialects.postgresql import JSONB, UUID
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import Query, Session, sessionmaker
+
+Metadata = MetaData()
+Base = declarative_base(metadata=Metadata)
+
+TABLE_INTERACTIONS = "interactions"
+
+
+class DataStoreInteraction(Base):
+    __tablename__ = TABLE_INTERACTIONS
+    interaction_id = Column(UUID, primary_key=True, default=str(uuid4()))
+    ts = Column(DateTime, default=datetime.now())
+    bot = Column(String, nullable=False)
+    data = Column(JSONB, default=dict())
+    message_channel = Column(String, default=None)
+    message_ts = Column(DateTime, default=None)
+    ttl = Column(Integer, default=None)
+    followup_ts = Column(DateTime, default=None)
+    followup_action = Column(String, default=None)
+    cron_followup_action = Column(
+        String, default=None
+    )  # TODO(zpriddy): Do I need? Is this the same as followup_action?
+    followed_up = Column(DateTime, default=None)
+    followed_up_ts = Column(DateTime, default=None)
+
+    @property
+    def __sql_values__(self) -> dict:
+        values = self.__dict__.copy()
+        values.pop("_sa_instance_state")
+        values.pop("interaction_id")
+        return values
+
+    @property
+    def object(self):
+        return self
+
+    def update_row(self, session: Session):
```

Comment:
Just update the attributes on the row instead of doing a query and then updating there rows.